code/renderers/solid/src/render.tsx

```diff
@@ -0,0 +1,127 @@
+import type { Component } from 'solid-js';
+import { ErrorBoundary, onMount } from 'solid-js';
+import { createStore } from 'solid-js/store';
+import { render as solidRender } from 'solid-js/web';
+import type { RenderContext, ArgsStoryFn, Args, StoryFn } from '@storybook/types';
+import type { SolidRenderer, StoryContext } from './types';
+
+/**
+ * SolidJS store for handling fine grained updates
+ * of the story args.
+ */
+const [store, setStore] = createStore({
+  args: {} as Args & StoryContext<SolidRenderer>,
+});
+
+let globals: StoryContext<SolidRenderer>['globals'];
+
+/**
+ * Checks when the story requires to be remounted.
+ * Elements outside the story requires a whole re-render.
+ * e.g. dark theme, show grid, etc...
+ */
+const remount = (force: boolean, context: StoryContext<SolidRenderer>) => {
+  let flag = false;
+
+  if (force) {
+    flag = true;
+  } else if (!Object.is(globals, context.globals)) {
+    // Globals refers to storybook visualization options.
+    globals = context.globals;
+    flag = true;
+  }
+  return flag;
+};
```

Comment:
1. Is the point here to force a re-mount when globals change?
2. Nit: I think the logic can be simplified a little bit.
```suggestion
/**
 * Checks when the story requires to be remounted.
 * Elements outside the story requires a whole re-render.
 * e.g. dark theme, show grid, etc...
 */
const shouldRemount = (force: boolean, context: StoryContext<SolidRenderer>) => {

  if (force) {
    return true;
  }
  
  if (!Object.is(globals, context.globals)) {
    // Globals refers to storybook visualization options.
    globals = context.globals;
    return true;
  }
  
  return false;
};
```

Reply:
1. Yes because fine-grained updates are not possible in React, so it needs to re-run the component to update the storybook canvas with globals.

Reply:
I don't think this is the "correct" behaviour. Globals should cause a re-render in the same way that args do.